frontend/components/MatchingUI/MatchContext.tsx

```diff
@@ -69,9 +81,37 @@ export const MatchContextProvider = ({
     // todo redirect to room
   };
 
-  const onChangeComplexity = (complexity: QuestionComplexity) => {
+  const onChangeComplexity = async (complexity: QuestionComplexity) => {
     setSelectComplexity(complexity);
     setMatchStatus(MatchStatus.MATCHING);
+    //TODO: adjust this
+    const ws = new WebSocket(
+      getPairingServiceUri({
+        userId: user.user!.email!,
+        complexity: selectedComplexity!,
+      }),
+    );
+    ws.onmessage = (msg) => {
+      try {
+        const parsed = JSON.parse(msg.data);
+        if (parsed.data.url) {
+          setEditorUri(parsed.data.url);
+          setMatchStatus(MatchStatus.MATCH_SUCCESS);
+          // TODO: replace toast with actual usage of returned details
+          toast.success(`Matched with ${JSON.stringify(parsed.data)}`);
+          ws.close();
+        } else if (parsed.status == 200) {
+          console.log(parsed);
```

Comment:
just to check, is there a use case for `! parsed.data.url` && `status = 200`  && not match error? is it just considered match error already?